force-app/repository/Repository.cls

```diff
@@ -187,6 +216,11 @@ public virtual without sharing class Repository implements IRepository {
     return this;
   }
 
+  public Set<String> getSelectFields() {
+    this.selectFields.addAll(this.addSelectfields());
+    return this.selectFields;
```

Comment:
I'm hesitant to expose this collection directly because that leaves it open to modification.

One potential solution is to copy the list - eg `return new Set<String>(this.select fields)` but I also wonder at the general value in the select fields being "viewable" outside the repository. I think the alternative would be for the "outer" repository to call the method being used within this method to collect the select fields - because we're still within the Repository class, calling a private method on another instance of the class is allowed.

What do you think? I am loathe to add another method to the interface unless it truly is necessary and will end up being used.